src/hazmat/lucas.rs

```diff
@@ -342,19 +340,22 @@ pub fn lucas_test<const L: usize>(
 
     // Some constants in Montgomery form
 
-    let params = MontyParams::<L>::new(*candidate);
+    let params = <T as Integer>::Monty::new_params(candidate.clone());
 
-    let zero = MontyForm::<L>::zero(params);
-    let one = MontyForm::<L>::one(params);
-    let two = one + one;
-    let minus_two = -two;
+    let zero = <T as Integer>::Monty::zero(params.clone());
+    let one = <T as Integer>::Monty::one(params.clone());
+    let two = one.clone() + &one;
+    let minus_two = -two.clone();
 
     // Convert Q to Montgomery form
 
     let q = if q_is_one {
-        one
+        one.clone()
     } else {
-        let abs_q = MontyForm::<L>::new(&Uint::<L>::from(abs_q), params);
+        let abs_q = <T as Integer>::Monty::new(
+            T::from(abs_q).widen(candidate.bits_precision()),
```

Comment:
When instantiating `BoxedMontyForm::new(residue, params)`, the current behavior is to panic when the residue's precision is not equal to the parameter's precision. Should this be changed to "automatic widening"?

Reply:
Sure, sounds good